src/net.cpp

```diff
@@ -1141,9 +1142,9 @@ void CConnman::InactivityCheck(CNode *pnode)
             LogPrintf("socket receive timeout: %is\n", nTime - pnode->nLastRecv);
             pnode->fDisconnect = true;
         }
-        else if (pnode->nPingNonceSent && pnode->nPingUsecStart + TIMEOUT_INTERVAL * 1000000 < GetTimeMicros())
+        else if (pnode->nPingNonceSent && pnode->m_ping_start.load() + std::chrono::seconds{TIMEOUT_INTERVAL} < GetTime<std::chrono::microseconds>())
```

Comment:
Not changed by this PR, but noting that this seems to be the only place in the codebase where `nPingNonceSent` is tested for truthiness with implicit conversion instead of against an integer value.

Reply:
Indeed unrelated.